lib/src/main/java/graphql/nadel/engine/plan/NadelExecutionPlanFactory.kt

```diff
@@ -60,60 +61,64 @@ internal class NadelExecutionPlanFactory(
         rootField: ExecutableNormalizedField,
         serviceHydrationDetails: ServiceExecutionHydrationDetails?,
     ): NadelExecutionPlan {
-        val executionSteps = mutableListOf<AnyNadelExecutionPlanStep>()
+        val executionSteps: MutableMap<ExecutableNormalizedField, List<NadelExecutionPlan.Step<Any>>> =
+            mutableMapOf()
 
         executionContext.timer.batch { timer ->
             traverseQuery(rootField) { field ->
-                transformsWithTimingStepInfo.forEach { transformWithTimingInfo ->
+                val steps = transformsWithTimingStepInfo.mapNotNull { transformWithTimingInfo ->
```

Comment:
Instead of adding to executionSteps as we go, we create a List of all the applicable transforms for this field, then set it into the executionSteps: Map